lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp

```diff
@@ -1078,6 +1092,14 @@ bool ProxyShape::doSelect(SelectionUndoHelper& helper, const SdfPathVector& orde
       uint32_t hasNodesToCreate = 0;
       for(auto prim : insertPrims)
       {
+        if(prim.IsPseudoRoot())
+        {
+          // For pseudo root, just modify maya's selection, don't alter
+          // our internal paths
+          newlySelectedPaths.append(MFnDagNode(thisMObject()).fullPathName());
```

Comment:
This feels a little dirty. It doesn't handle instancing, not that I have a better suggestion right now...

Reply:
Agreed, this made my skin itch too.  However, (as you also pointed out), it's far from the first place in the code base where we're getting a dag path from only an MObject - for instance, in the same file:

https://github.com/AnimalLogic/AL_USDMaya/blob/develop/lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp#L518

...and trying to get the plugin as a whole to behave well with instances seemed a much larger scope than this small fix.

However... I can also understand not wanting to make a problem worse.  Unfortunately, the only way I can think of to do this "right" would be to modify the signature of "doSelect" to take an additional argument - a dagPath (or at least an instance index... though I think a dagPath would be preferable) for the particular proxyShape instance we're selection.  Then, we'd also need to modify ProxyShapeCommandBase::getShapeNode() to return a dag path, not a ProxyShape*.

That seemed like a wider-reaching change than I wanted for this bugfix PR... but if you're on board with it, I could do that chunk, which would be a first small step towards instance compatibility...